test/float8/test_everything_single_gpu.sh

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+#!/bin/bash
+
+# terminate script on first error
+set -e
+
+pytest test/float8/test_base.py --verbose -s
+pytest test/float8/test_compile.py --verbose -s
+pytest test/float8/test_numerics_integration.py --verbose -s
+pytest test/float8/test_auto_filter.py --verbose -s
+pytest test/integration --verbose -s
```

Comment:
nit: These are not specific to float8, but I wanted to amke sure they were run on h100 node. It feels a little strange to have in thsi file, can we just keep all the tests closer to the workflow so when people wnat to expand h100 tests they can update.

Also the job name should be changed in retrospect to h100 tests or whatever

Reply:
Hmm, I see. Yeah I guess we can split workflows based on hardware type, that would also make scheduling/allocation faster if all tests that need 1xH100 share that job, all tests that need 4xH100 share that job, etc.

Reply:
splitting tests by product and CI jobs by hardware type makes sense to me

Reply:
Split the CI workflows into:
- 1xH100
- 4xH100
- 1xL4 (SM89)

(We could also name them SM89/SM90)

This is ready for another look.